internal/build/docker_builder.go

```diff
@@ -68,6 +69,31 @@ func (d *dockerImageBuilder) BuildImage(ctx context.Context, ps *PipelineState,
 	return d.buildFromDf(ctx, ps, db, paths, filter, refs)
 }
 
+func (d *dockerImageBuilder) DumpImageDeployRef(ctx context.Context, ref string) (reference.NamedTagged, error) {
+	refParsed, err := container.ParseNamed(ref)
+	if err != nil {
+		return nil, errors.Wrap(err, "DumpImageDeployRef")
+	}
+
+	data, _, err := d.dCli.ImageInspectWithRaw(ctx, ref)
```

Comment:
Have a vague instinct that as much of this logic as possible should live in a helper function that's called by both `dump-image...` and the `CustomBuilder`, to guard against the logic going out of sync. How hard would that be?

Reply:
whoops yes. cool!